core-web/libs/sdk/client/src/lib/client/page/page-api.ts

```diff
@@ -0,0 +1,371 @@
+import { buildPageQuery, buildQuery, fetchGraphQL, mapResponseData } from './utils';
+
+import { DotCMSClientConfig, RequestOptions } from '../client';
+import { ErrorMessages } from '../models';
+import { DotCMSGraphQLPageResponse, DotCMSPageAsset } from '../models/types';
+
+/**
+ * The parameters for the Page API.
+ * @public
+ */
+export interface PageRequestParams {
+    /**
+     * The id of the site you want to interact with. Defaults to the one from the config if not provided.
+     */
+    siteId?: string;
+
+    /**
+     * The mode of the page you want to retrieve. Defaults to the site's default mode if not provided.
+     */
+    mode?: 'EDIT_MODE' | 'PREVIEW_MODE' | 'LIVE';
+
+    /**
+     * The language id of the page you want to retrieve. Defaults to the site's default language if not provided.
+     */
+    languageId?: number | string;
+
+    /**
+     * The id of the persona for which you want to retrieve the page.
+     */
+    personaId?: string;
+
+    /**
+     * Whether to fire the rules set on the page.
+     */
+    fireRules?: boolean | string;
+
+    /**
+     * Allows access to related content via the Relationship fields of contentlets on a Page; 0 (default).
+     */
+    depth?: 0 | 1 | 2 | 3 | '0' | '1' | '2' | '3';
+
+    /**
+     * The publish date of the page you want to retrieve.
+     */
+    publishDate?: string;
+}
+
+/**
+ * The private parameters for the Page API.
+ * @internal
+ */
+export interface BackendPageParams {
+    /**
+     * The id of the site you want to interact with.
+     */
+    hostId?: string;
+
+    /**
+     * The mode of the page you want to retrieve.
+     */
+    mode?: string;
+
+    /**
+     * The id of the persona for which you want to retrieve the page.
+     */
+    'com.dotmarketing.persona.id'?: string;
+
+    /**
+     * The language id of the page you want to retrieve.
+     */
+    language_id?: string;
+
+    /**
+     * Whether to fire the rules set on the page.
+     */
+    fireRules?: string;
+
+    /**
+     * The depth of the page you want to retrieve.
+     */
+    depth?: string;
+
+    /**
+     * The publish date of the page you want to retrieve.
+     */
+    publishDate?: string;
+}
```

Comment:
```suggestion
/**
 * Maps frontend parameter types to backend string types
 */
type StringifyParam<T> = T extends string | number | boolean ? string : never;

/**
 * The private parameters for the Page API.
 * @internal
 */
export type BackendPageParams = {
    [K in keyof PageRequestParams as K extends 'siteId'
        ? 'hostId'
        : K extends 'languageId'
        ? 'language_id'
        : K extends 'personaId'
        ? 'com.dotmarketing.persona.id'
        : K]?: StringifyParam<PageRequestParams[K]>;
};
```

Of course I ai this type, I'm not †hat smart, but what I'm looking with this is keep the relationship of the two types.

What this is doing is creating a new type `BackendPageParams` by iterating trough the `PageRequestParams` and changing some keys based on the mapping.

`siteId --> hostId` and so on...

And the `StringifyParam<PageRequestParams[K]>`: Converts the value types to `string` to send to the backend.

Is just a suggestions, but we don't have to use it... I just think the relationship here is important.

Reply:
Good catch, i added a mapping for this